custom/icds_reports/views.py

```diff
@@ -2065,20 +2067,28 @@ def post(self, request, *args, **kwargs):
         year = int(request.POST.get('year', None))
         selected_date = date(year, month, 1).strftime('%Y-%m-%d')
 
-        if state_id and not user_can_access_location_id(
+        if location_id and not user_can_access_location_id(
                 self.kwargs['domain'], request.couch_user, state_id
         ):
             return JsonResponse({"message": "Sorry, you do not have access to that location."})
 
-        sync, _ = get_cas_data_blob_file(data_type, state_id, selected_date)
+        location = SQLLocation.objects.get(location_id=location_id)
```

Comment:
nit: `select_related` on the `location_type` i think would avoid a second db query.